### PR TITLE
Add initial support for github organizations

### PR DESCRIPTION
**Pull request by ficristo**
This should hopefully fix https://github.com/adobe/brackets/issues/12464 and help https://github.com/adobe/brackets/issues/8228.
ATM you cannot upload as organization the first time. Before you have to change the owner to an organization and only after the members can upload new versions.
If there is still time for 1.8 would be a nice to have.
Do I need to do some changes to brackets too?

### CODE REVIEW

File: lib/user_utils.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+/*jslint vars: true, plusplus: true, nomen: true, node: true, indent: 4, maxerr: 50 */
+
+"use strict";
+
+function isUserOwner(entry, user) {
+    if (entry.owner === user.owner) {
+        return true;
+    }
+
+    if (entry.owner === user.owner) {
```
Comment (ingorichter) [inline lib/user_utils.js]: one too many :-)

File: lib/user_utils.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        return true;
+    }
+
+    return false;
+}
+
+function isUserOrgOwner(entry, user) {
+    var i;
+    for (i = 0; i < user.orgs.length; i++) {
```
Comment (zaggino) [inline lib/user_utils.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some ?

File: lib/user_utils.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+/*jslint vars: true, plusplus: true, nomen: true, node: true, indent: 4, maxerr: 50 */
+
+"use strict";
+
+function isUserOwner(entry, user) {
+    if (entry.owner === user.owner) {
```
Comment (zaggino) [inline lib/user_utils.js]: just `return entry.owner === user.owner` would seem enough

File: lib/user_utils.js (hunk 2 above)
Comment (ficristo) [inline lib/user_utils.js]: Since was already in the tree I used lodash.

**Comment by zaggino** [comment]
Looks good, I don't believe we need to change anything in Brackets itself

**Comment by ficristo** [comment]
In the source code there is written that `lib\registry_utils.js` should be kept in sync with the one in Brackets `src\extensibility\registry_utils.js`.
But it seems to me that `lastVersionDate` and `authorInfo` are different and `formatDownloadURL` is not present in the Brackets file.
What should I do?

**Comment by zaggino** [comment]
Best would be to diff the files and make them equal.

**Comment by zaggino** [comment]
From my quick check, they look pretty much the same except the AMD wrapper

**Comment by ingorichter** [comment]
When get this merged, we still need to find somebody who has access to the machine that hosts the registry. Somebody took away my permissions and I no longer can update the app itself.

**Comment by zaggino** [comment]
:cry:

**Comment by swmitra** [comment]
@ficristo The changes looks good to me. I am merging this PR and once we get permission to update app code, I will notify all committers to check the registry app for corresponding updates.

**Comment by Hirse** [comment]
@ficristo, @swmitra Thanks for looking into this.
Is there any update? Or can you make a guess when this will be usable?
Eagerly awaiting the feature for the Brackets Beautify extension.

**Comment by ficristo** [comment]
Unfortunately production doesn't have this change yet.
@swmitra or @saurabh95 maybe have more info.

**Comment by saurabh95** [comment]
@Hirse  We are waiting for access to the brackets-registry in the prod server, once we get it we will update the prod. We will try to have this update along with 1.9 release.

**Comment by saurabh95** [comment]
@Hirse We have merged this PR in production, Could you check it now?
If it is not working,  you may have to sign-out and sign-in again.

**Comment by Hirse** [comment]
Yes, seems to work. 
Great work, thanks guys.

**Pull request merged**